pkg/controller/statefulset/stateful_update_utils.go

```diff
@@ -0,0 +1,47 @@
+package statefulset
+
+import (
+	appsv1alpha1 "github.com/openkruise/kruise/pkg/apis/apps/v1alpha1"
+	"github.com/openkruise/kruise/pkg/util/priorityupdate"
+	v1 "k8s.io/api/core/v1"
+)
+
+func sortPodsToUpdate(rollingUpdateStrategy *appsv1alpha1.RollingUpdateStatefulSetStrategy, updateRevision string, replicas []*v1.Pod) []int {
+	var updateMin int
+	var priorityStrategy *appsv1alpha1.UpdatePriorityStrategy
+	if rollingUpdateStrategy != nil {
+		if rollingUpdateStrategy.Partition != nil {
+			updateMin = int(*rollingUpdateStrategy.Partition)
+		}
+		if rollingUpdateStrategy.UnorderedUpdate != nil {
+			priorityStrategy = rollingUpdateStrategy.UnorderedUpdate.PriorityStrategy
+		}
+	}
+	maxUpdate := len(replicas) - updateMin
+	if maxUpdate <= 0 {
+		return []int{}
+	}
+
+	var updatedIdxs []int
+	var waitUpdateIdxs []int
+	for target := len(replicas) - 1; target >= 0; target-- {
+		if isTerminating(replicas[target]) {
+			updatedIdxs = append(updatedIdxs, target)
+		} else if getPodRevision(replicas[target]) == updateRevision {
+			updatedIdxs = append(updatedIdxs, target)
+		} else {
+			waitUpdateIdxs = append(waitUpdateIdxs, target)
+		}
+	}
+
+	if priorityStrategy != nil {
+		waitUpdateIdxs = priorityupdate.SortIndexesByPriority(priorityStrategy, replicas, waitUpdateIdxs)
+	}
+
+	allIdxs := append(updatedIdxs, waitUpdateIdxs...)
+	if len(allIdxs) > maxUpdate {
+		allIdxs = allIdxs[:maxUpdate]
```

Comment:
Should this method return To-be-updated idx only?

Reply:
It can not. The update loop in `stateful_set_control.go` requires all pods both updated and prepared to update.